File: L2JTeon-dp/data/scripts/ai/group_template/cats_eye_bandit.py
import sys
from net.sf.l2j.util import Rnd
from net.sf.l2j.gameserver.model.quest.jython import QuestJython as JQuest
from net.sf.l2j.gameserver.serverpackets import NpcSay
from net.sf.l2j.gameserver.serverpackets import PlaySound

# Cats_Eye_Bandit
class cats_eye_bandit(JQuest) :

    # init function.  Add in here variables that you'd like to be inherited by subclasses (if any)
    def __init__(self,id,name,descr):
        self.cats_eye_bandit = 27038
        self.FirstAttacked = False
        # finally, don't forget to call the parent constructor to prepare the event triggering
        # mechanisms etc.
        JQuest.__init__(self,id,name,descr)

    def onAttack (self,npc,player,damage,isPet):
        objId=npc.getObjectId()
        if self.FirstAttacked:
           if Rnd.get(40) : return
           npc.broadcastPacket(NpcSay(objId,0,npc.getNpcId(),"You childish fool, do you think you can catch me?"))
        else :
           self.FirstAttacked = True
        return

    def onKill (self,npc,player,isPet):
        npcId = npc.getNpcId()
        if npcId == self.cats_eye_bandit:
            objId=npc.getObjectId()
            if Rnd.get(80) : npc.broadcastPacket(NpcSay(objId,0,npc.getNpcId(),"I must do something about this shameful incident..."))

            self.FirstAttacked = False
        elif self.FirstAttacked :
            self.addSpawn(npcId, npc.getX(), npc.getY(), npc.getZ())
        return

QUEST      = cats_eye_bandit(-1,"cats_eye_bandit","ai")

QUEST.addKillId(QUEST.cats_eye_bandit)

QUEST.addAttackId(QUEST.cats_eye_bandit)

File: trunk/L2JTeon-dp/data/scripts/ai/group_template/CatsEyeBandit.java
package ai.group_template;

import net.sf.l2j.gameserver.model.actor.instance.L2NpcInstance;
import net.sf.l2j.gameserver.model.actor.instance.L2PcInstance;
import net.sf.l2j.gameserver.serverpackets.NpcSay;
import net.sf.l2j.util.Rnd;

/**
 * @author Maxi
 */
public class CatsEyeBandit extends L2AttackableAIScript
{
	private static final int BANDIT = 27038;

	private static boolean _FirstAttacked;

	public CatsEyeBandit(int questId, String name, String descr)
	{
		super(questId, name, descr);
		int[] mobs = {BANDIT};
		registerMobs(mobs);
		_FirstAttacked = false;
	}

	public String onAttack (L2NpcInstance npc, L2PcInstance attacker, int damage, boolean isPet)
	{
        if (npc.getNpcId() == BANDIT)
        {
            if (_FirstAttacked)
            {
               if (Rnd.get(100) == 40)
            	   npc.broadcastPacket(new NpcSay(npc.getObjectId(),0,npc.getNpcId(),"You childish fool, do you think you can catch me?"));
            }
            else
            {
               _FirstAttacked = true;
		}
        }
        return super.onAttack(npc, attacker, damage, isPet);
    }

    public String onKill (L2NpcInstance npc, L2PcInstance killer, boolean isPet)
	{
        int npcId = npc.getNpcId();
        if (npcId == BANDIT)
        {
		int objId = npc.getObjectId();
               if (Rnd.get(100) == 80)
            npc.broadcastPacket(new NpcSay(objId,0,npcId,"I must do something about this shameful incident..."));
            _FirstAttacked = false;
        }
        return super.onKill(npc,killer,isPet);
    }

	public static void main(String[] args)
	{
		new CatsEyeBandit(-1, "CatsEyeBandit", "ai");
	}
}